compiler/sem/pragmas.nim

```diff
@@ -1438,119 +1299,53 @@ proc prepareSinglePragma(
+          #       it, PAstDiag(kind: adSemCustomUserError, errmsg: s.strVal))
+          discard
       of wPragma:
-        if not sym.isNil and sym.kind == skTemplate:
-          sym.flags.incl sfCustomPragma
-        else:
-          result = processPragma(c, n, i)
-          result.flags.incl nfImplicitPragma # xxx: legacy singlepragma=true
+        result = noVal(c, it)
+        sym.flags.incl sfCustomPragma
       of wDiscardable:
         result = noVal(c, it)
-        if sym != nil: incl(sym.flags, sfDiscardable)
+        incl(sym.flags, sfDiscardable)
       of wNoInit:
         result = noVal(c, it)
-        if sym != nil: incl(sym.flags, sfNoInit)
+        incl(sym.flags, sfNoInit)
       of wCodegenDecl:
         result = processCodegenDecl(c, it, sym)
-      of wChecks, wObjChecks, wFieldChecks, wRangeChecks, wBoundChecks,
-         wOverflowChecks, wNilChecks, wAssertions, wWarnings, wHints,
-         wLineDir, wOptimization, wStaticBoundchecks, wStyleChecks,
-         wCallconv, wDebugger, wProfiler,
-         wFloatChecks, wNanChecks, wInfChecks, wTrMacros:
-        var tmp = c.config.options
-        result = processOption(c, it, tmp)
-        c.config.options = tmp
       of wStackTrace, wLineTrace:
-        if sym.kind in {skProc, skMethod, skConverter}:
-          result = processOption(c, it, sym.options)
-        else:
-          var tmp = c.config.options
-          result = processOption(c, it, tmp)
-          c.config.options = tmp
+        # TODO: also support for funcs and (closure) iterators -- there's no
+        #       reason not to
+        assert sym.kind in {skProc, skMethod, skConverter}
+        result = processOption(c, it, sym.options)
       of FirstCallConv..LastCallConv:
         assert(sym != nil)
+        assert sym.typ != nil
         result = it
-        if sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
-          sym.typ.callConv = wordToCallConv(k)
-          sym.typ.flags.incl tfExplicitCallConv
-      of wEmit:
-        result = pragmaEmit(c, it)
-      of wLinearScanEnd, wComputedGoto:
-        result = noVal(c, it)
+        sym.typ.callConv = wordToCallConv(k)
+        sym.typ.flags.incl tfExplicitCallConv
       of wEffects:
         # is later processed in effect analysis:
         result = noVal(c, it)
       of wIncompleteStruct:
         result = noVal(c, it)
-        if sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
-          incl(sym.typ.flags, tfIncompleteStruct)
+        assert sym.typ != nil
+        incl(sym.typ.flags, tfIncompleteStruct)
       of wCompleteStruct:
         result = noVal(c, it)
-        if sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
-          incl(sym.typ.flags, tfCompleteStruct)
+        assert sym.typ != nil
+        incl(sym.typ.flags, tfCompleteStruct)
       of wUnchecked:
+        # XXX: seems redundant with ``tyUncheckedArray``. Remove it?
```

Comment:
Yup, it was supposedly removed in Nim 0.20.0 (per the change log).

Reply:
Removed the pragma and special word.